dataline-tests/src/acceptanceTests/java/io/dataline/test/acceptance/AcceptanceTests.java

```diff
@@ -75,61 +82,169 @@
-  public static void init() throws IOException, InterruptedException {
-    SOURCE_PSQL = new PostgreSQLContainer();
-    TARGET_PSQL = new PostgreSQLContainer();
-    SOURCE_PSQL.start();
-    TARGET_PSQL.start();
+  private List<UUID> sourceImplIds = Lists.newArrayList();
+  private List<UUID> connectionIds = Lists.newArrayList();
+
+  @BeforeEach
+  public void init() throws IOException, InterruptedException {
+    sourcePsql = new PostgreSQLContainer();
+    targetPsql = new PostgreSQLContainer();
+    sourcePsql.start();
+    targetPsql.start();
+
+    PostgreSQLContainerHelper.runSqlScript(MountableFile.forClasspathResource("simple_postgres_init.sql"), sourcePsql);
+  }
+
+  @AfterEach
+  public void tearDown() throws ApiException {
+    sourcePsql.stop();
+    targetPsql.stop();
+
+    for (UUID sourceImplId : sourceImplIds) {
+      deleteSourceImpl(sourceImplId);
+    }
+
+    for (UUID connectionId : connectionIds) {
+      disableConnection(connectionId);
+    }
+
+    // TODO disable destination once the API exposes the functionality.
+  }
+
+  @Test
+  public void testCreateDestinationImpl() throws ApiException {
+    Map<Object, Object> destinationDbConfig = getDestinationDbConfig();
+    UUID postgresDestinationSpecId = getPostgresDestinationSpecId();
+    UUID workspaceId = PersistenceConstants.DEFAULT_WORKSPACE_ID;
+    String name = "AccTestDestinationDb-" + UUID.randomUUID().toString();
+
+    DestinationImplementationRead destinationImpl = createDestinationImplementation(
+        name,
+        workspaceId,
+        postgresDestinationSpecId,
+        getDestinationDbConfig());
+
+    assertEquals(name, destinationImpl.getName());
+    assertEquals(postgresDestinationSpecId, destinationImpl.getDestinationSpecificationId());
+    assertEquals(workspaceId, destinationImpl.getWorkspaceId());
+    assertEquals(destinationDbConfig, destinationImpl.getConnectionConfiguration());
+  }
+
+  @Test
+  public void testDestinationCheckConnection() throws ApiException {
+    UUID destinationImplId = createPostgresDestinationImpl().getDestinationImplementationId();
+    CheckConnectionRead.StatusEnum checkOperationStatus = apiClient.getDestinationImplementationApi()
+        .checkConnectionToDestinationImplementation(
+            new DestinationImplementationIdRequestBody().destinationImplementationId(destinationImplId))
+        .getStatus();
+
+    assertEquals(CheckConnectionRead.StatusEnum.SUCCESS, checkOperationStatus);
+  }
 
-    PostgreSQLContainerHelper.runSqlScript(MountableFile.forClasspathResource("simple_postgres_init.sql"), SOURCE_PSQL);
+  @Test
+  public void testCreateSourceImplementation() throws ApiException {
+    String dbName = "acc-test-db";
+    UUID postgresSourceSpecId = getPostgresSourceSpecId();
+    UUID defaultWorkspaceId = PersistenceConstants.DEFAULT_WORKSPACE_ID;
+    Map<Object, Object> sourceDbConfig = getSourceDbConfig();
+    SourceImplementationRead response = createSourceImplementation(
```

Comment:
Nit new line for consistency with the previous test

Reply:
done